addons/controllers/clusterbootstrap_controller.go

```diff
@@ -902,6 +927,41 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecretOnRemote(c
 		for k, v := range patchedSecret.Data {
 			remoteSecret.Data[k] = v
 		}
+		// Add TKR NodeSelector info if it's a TKGS cluster
+		infraRef, err := util.GetInfraProvider(cluster)
+		if err != nil {
+			return err
+		}
+		if infraRef == tkgconstants.InfrastructureProviderVSphere {
+			ok, err := util.IsTKGSCluster(r.context, r.Client, cluster)
+			if err != nil {
+				return err
+			}
+			if ok {
+				upgradeDataValues := addontypes.UpgradeDataValues{
```

Comment:
I would rename the var and struct to TKGSDataValues since these are used for initial install as well.

Reply:
renamed